modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheConcurrentMapImpl.java

```diff
@@ -95,7 +95,7 @@ protected final GridCacheMapEntry putEntryIfObsoleteOrAbsent(
         GridCacheMapEntry doomed = null;
 
         boolean done = false;
-        boolean reserved = false;
+        boolean reserved = clearing;
```

Comment:
Let's rename this flag to skipReserve as this is the only semantics of this flag. Also, we need to get rid of the `touch` flag in the parent call hierarchy as the naming is confusing.